src/libraries/System.Private.CoreLib/src/System/Resources/ResourceReader.Core.cs

```diff
@@ -65,24 +67,47 @@ private object DeserializeObject(int typeIndex)
             return graph;
         }
 
+        // TODO: Remove this DynamicDependencyAttributes when https://github.com/mono/linker/issues/943 is fixed.
+        [DynamicDependency("Deserialize", "System.Runtime.Serialization.Formatters.Binary.BinaryFormatter", "System.Runtime.Serialization.Formatters")]
```

Comment:
Note that today the `BinaryFormatter` type exists, and is empty in a default Blazor app. So this is going to make the app size larger.

We should implement a way to turn this behavior off. There are probably 2 levels that both could be done.

1. Completely removing BinaryFormatter from an app, no matter where it is used. This was discussed in the [feature-switch.md#security](https://github.com/dotnet/designs/blob/master/accepted/2020/feature-switch.md#security) section of the design doc.
2. Disallow deserialization in ResourceReader. The boolean is already there, we would just need to add a feature switch for it:
https://github.com/dotnet/runtime/blob/38f8e3808db0dca7605f294fd6d2d024755997fe/src/libraries/System.Private.CoreLib/src/System/Resources/ResourceReader.Core.cs#L47-L50

See the discussion here https://github.com/dotnet/runtime/issues/32862#issuecomment-599837318

cc @joperezr since he is already working in the Resources area for linking.

Reply:
On a second thought, I should probably just remove this part of the change - if the BinarySerializer indeed gets used for something, after trimming, chances are pretty low that we would end up in a happy place. I'll instead tag this and a couple other places as RequiresUnreferencedCode to get rid of some of the warning noise and leave this for the feature switches to take care of.